common/http/src/retry.rs

```diff
@@ -110,13 +111,28 @@ impl Retry {
                     .and_then(|s| FromStr::from_str(s).ok())
                     .ok_or_else(|| Error::InvalidFileName(file_name.clone()))?;
 
+                let file_size = match metadata(path.clone()).await {
+                    Ok(md) => md.len(),
+                    Err(e) => {
+                        warn!(
+                            "retry file size unknown, metrics may be skewed; reason={}",
+                            e
+                        );
+                        0
+                    }
+                };
+
+                retry_disk_used += file_size;
+
                 if Utc::now().timestamp() - timestamp < self.retry_base_delay_secs {
                     continue;
                 }
+                Metrics::retry().inc_pending();
                 self.waiting.push(path);
             }
         }
 
+        Metrics::retry().report_storage_used(retry_disk_used);
```

Comment:
Instead of updating the gauge when the file system is polled, would it be more dynamic to call `add()` and `sub()` on the gauge every time a file is added / removed from the file system?

Reply:
@jorgebay I initially had coded the implementation that way but ran into two issues with that implementation:
1. There are multiple paths in our code where a retry file may be removed. When a retry block is sent again but fails, the retry code writes out a new retry file and cleans up the prior version. This was causing double counting of disk space. 
2. If the agent should crash with files in the retry directory, I wanted to ensure that the agent emits accurate disk stats on restart.

In both cases, it seemed easier to compute the full disk usage metric and allow the metric to correct itself in case we miss a retry cleanup or if the agent is trying to recover after a crash.

Reply:
Sounds good, in any case, it's a great improvement to have the size of the queue in the metrics 👍 
We can revisit in the future in case the gauge shows stale data.